src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -604,8 +604,14 @@ public static string GetMsBuildPathFromVSDirectory(DirectoryInfo vsDirectory, Li
                     {
                         var folderName = GetVersionFolder(msbuild.FullName);
                         if (folderName.ToLower() == "current") return -100;
-                        else return -Convert.ToDouble(folderName);
-
+                        else
+                        {
+                            if(double.TryParse(folderName, out double result))
```

Comment:
this logic makes sense. But in my case I do have current version of VS2022, which won't even enter else block it will return value at line 606

Reply:
Are you codelyzer tests failing, or is this an issue on the client?

Reply:
it's the issue on the client